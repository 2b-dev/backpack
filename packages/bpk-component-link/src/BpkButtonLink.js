import React, { PropTypes } from 'react';

import './bpk-link.scss';

const BpkButtonLink = (props) => {
  const classNames = ['bpk-link'];

  if (props.white) { classNames.push('bpk-link--white'); }

  return (
    <button
      type="button"
      className={classNames.join(' ')}
      onClick={props.onClick}
    >
      {props.children}
    </button>
  );
};

BpkButtonLink.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  white: PropTypes.bool,
};

BpkButtonLink.defaultProps = {
  white: false,
};

export default BpkButtonLink;
