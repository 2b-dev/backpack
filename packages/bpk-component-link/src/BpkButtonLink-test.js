import React from 'react'
import renderer from 'react/lib/ReactTestRenderer'
import BpkButtonLink from './BpkButtonLink'

describe('BpkButtonLink', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<BpkButtonLink onClick={() => null}>Link</BpkButtonLink>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
