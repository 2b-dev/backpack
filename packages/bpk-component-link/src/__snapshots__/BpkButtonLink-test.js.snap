exports[`BpkButtonLink should render correctly 1`] = `
<button
  className="bpk-link"
  onClick={[Function]}
  type="button">
  Link
</button>
`;

exports[`BpkButtonLink should render correctly with "white" attribute 1`] = `
<button
  className="bpk-link bpk-link--white"
  onClick={[Function]}
  type="button">
  Link
</button>
`;
